cmd/runhcs/container.go

```diff
@@ -336,8 +336,8 @@ func createContainer(cfg *containerConfig) (_ *container, err error) {
 		}
 
 		// Determine the network namespace to use.
-		if cfg.Spec.Windows.Network != nil && cfg.Spec.Windows.Network.NetworkSharedContainerName != "" {
-			err = stateKey.Get(cfg.Spec.Windows.Network.NetworkSharedContainerName, keyNetNS, &netNS)
+		if cfg.Spec.Windows.Network != nil && cfg.Spec.Windows.Network.NetworkNamespace != "" {
```

Comment:
Do we need to support both?

Reply:
I see. Basically if its a v1 container we need to use this method. If its a v2 container we need to just use the NetworkNamespace value. In both cases We need to set `netNS` to this value which gets put on the `RequestedNetNS` of the persisted state. Then later this gets passed to the `hcsoci.CreateOptions` `NetworkNamespace`.

Reply:
I wasn't sure why NetworkSharedContainerName  was used in the first place to assign it to netNs. for V1, it would only have the primary or infra container Id.

Reply:
No its not that. Its the value of the primary infra container name. So we use the name to lookup the key (this is the ID of the infra container) and then the value stored at that key is the NetNS. This is how V1 worked yes?

Reply:
I need some more understand of what we are trying to store in registry. How do I set this key up, so that Get wouldn't fail. Will sync up offline